oresat_configs/scripts/gen_xtce.py

```diff
@@ -142,247 +152,635 @@ def write_xtce(config: OreSatConfig, dir_path: str = ".") -> None:
+                        )
+                    ET.SubElement(
+                        arg_list,
+                        "Argument",
+                        attrib={
+                            "name": name,
+                            "argumentTypeRef": type_name,
+                        },
+                    )
+                    ET.SubElement(
+                        cmd_entry_list,
+                        "ArgumentRefEntry",
+                        attrib={"argumentRef": name},
+                    )
+
+                if req_field.name in ["opd_addr", "node_id"]:
+                    type_name = req_field.name + "_type"
+                else:
+                    type_name = req_field.data_type + "_type"
+
+                if type_name not in arg_types:
+                    arg_types.append(type_name)
+                    _add_argument_type(arg_type_set, req_field, type_name)
                 ET.SubElement(
-                    enum_list,
-                    "Enumeration",
+                    arg_list,
+                    "Argument",
                     attrib={
-                        "value": str(value),
-                        "label": name,
+                        "name": req_field.name,
+                        "argumentTypeRef": type_name,
                     },
                 )
-        elif obj.data_type in canopen.objectdictionary.INTEGER_TYPES:
-            if obj.data_type in canopen.objectdictionary.UNSIGNED_TYPES:
-                signed = False
-                encoding = "unsigned"
-            else:
-                signed = True
-                encoding = "twosComplement"
-
-            para_type = ET.SubElement(
-                tm_meta_para,
-                "IntegerParameterType",
-                attrib={
-                    "name": name,
-                    "signed": str(signed).lower(),
-                },
-            )
-
-            para_unit_set = ET.SubElement(para_type, "UnitSet")
-            if obj.unit:
-                para_unit = ET.SubElement(
-                    para_unit_set,
-                    "Unit",
-                    attrib={
-                        "description": obj.unit,
-                    },
+                ET.SubElement(
+                    cmd_entry_list,
+                    "ArgumentRefEntry",
+                    attrib={"argumentRef": req_field.name},
                 )
-                para_unit.text = obj.unit
 
-            data_enc = ET.SubElement(
-                para_type,
-                "IntegerDataEncoding",
+        ET.SubElement(
+            cmd_entry_list,
+            "FixedValueEntry",
+            attrib={"name": "hmac", "binaryValue": "0" * 64, "sizeInBits": str(32 * 8)},
+        )
+        ET.SubElement(
+            cmd_entry_list,
+            "FixedValueEntry",
+            attrib={"name": "uslp_fecf", "binaryValue": "0000", "sizeInBits": "16"},
+        )
```

Comment:
xtcetools' [BogusSAT-1](https://gitlab.com/dovereem/xtcetools/-/blob/master/src/test/resources/org/xtce/toolkit/test/BogusSAT-1.xml) has an example of how to do a packet footer with `BaseContainer`. Apparently you need to set `LocationInContainerInBits referenceLocation="containerEnd"` on the `ParameterRefEntry`s that you want at the end of a container, and then it'll arrange for those fields to come after all the fields in anything that uses the container as a `BaseContainer`. I'm unclear if there's other tags that will take `referenceLocation` or if you only have bits to specify.

But so yeah, `hmac` and `uslp_fecf` can be moved into `uslp_header` (possibly renamed to just `uslp`).

Reply:
This only worked for parameters, not arguments